src/helpers.rs

```diff
@@ -169,9 +169,7 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let this = self.eval_context_mut();
         let param_env = ty::ParamEnv::reveal_all(); // in Miri this is always the param_env we use... and this.param_env is private.
         let callee_abi = f.ty(*this.tcx, param_env).fn_sig(*this.tcx).abi();
-        if callee_abi != caller_abi {
-            throw_ub_format!("calling a function with ABI {} using caller ABI {}", callee_abi.name(), caller_abi.name())
-        }
+        check_abi(caller_abi, callee_abi)?;
```

Comment:
This is not necessary to fix the message -- I just came across this when grepping for "calling a function with ABI...".

Reply:
This is conceptually a different error -- here Miri calls a closure, so the caller ABI is fixed and the error is that the callee ABI is wrong.

For the other errors, it's the other way around: the callee ABI is fixed and the caller ABI is wrong.

So these should not use the same codepath. Maybe the error messages should be slightly adjusted to emphasize which part is wrong.

Reply:
That makes sense. I reverted this part.